File: datasets/java/HumanEval_82_prime_length.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // Write a function that takes a string and returns True if the string
    // length is a prime number or False otherwise
    // Examples
    // prime_length('Hello') == True
    // prime_length('abcdcba') == True
    // prime_length('kittens') == True
    // prime_length('orange') == False
    public static boolean prime_length(String string) {
        int len = string.length();
        if (len == 0) {
            return false;
        }
        if (len == 1) {
            return true;
        }
        if (len == 2) {
            return true;
        }
        if (len % 2 == 0) {
            return false;
        }
        for (int i = 3; i < len; i += 2) {
            if (len % i == 0) {
                return false;
            }
        }
        return true;
    }
    // Write a function that takes a string and returns True if the string
    // is a palindrome or False otherwise
    // Examples
    // palindrome('Hello') == False
    // palindrome('abcdcba') == True
    // palindrome('kittens') == False
    // palindrome('orange') == False
    public static boolean palindrome(String string) {
        int len = string.length();
        if (len == 0) {
            return true;
        }
        if (len == 1) {
            return true;
        }
        if (len == 2) {
            return string.charAt(0) == string.charAt(1);
        }
        for (int i = 0; i < len / 2; i++) {
            if (string.charAt(i) != string.charAt(len - i - 1)) {
                return false;
            }
        }
        return true;
    }
    // Write a function that takes a string and returns True if the string
    // is a palindrome or False otherwise
    // Examples
    // palindrome('Hello') == False
    // palindrome('abcdcba') == True
    // palindrome('kittens') == False
    // palindrome('orange') == False
    public static boolean palindrome(String string) {
        int len = string.length();
        if (len == 0) {
            return true;
        }
        if (len == 1) {
            return true;
        }
        if (len == 2) {
            return

    }
    public static void main(String[] args) {
    assert(prime_length(("Hello")) == (true));
    assert(prime_length(("abcdcba")) == (true));
    assert(prime_length(("kittens")) == (true));
    assert(prime_length(("orange")) == (false));
    assert(prime_length(("wow")) == (true));
    assert(prime_length(("world")) == (true));
    assert(prime_length(("MadaM")) == (true));
    assert(prime_length(("Wow")) == (true));
    assert(prime_length(("")) == (false));
    assert(prime_length(("HI")) == (true));
    assert(prime_length(("go")) == (true));
    assert(prime_length(("gogo")) == (false));
    assert(prime_length(("aaaaaaaaaaaaaaa")) == (false));
    assert(prime_length(("Madam")) == (true));
    assert(prime_length(("M")) == (false));
    assert(prime_length(("0")) == (false));
    }

}

File: datasets/scala-keep-code_davinci_001_temp_0.2/HumanEval_82_prime_length.scala
import scala.math._
import scala.collection.mutable._
object Problem {
    // Write a function that takes a string and returns True if the string
    // length is a prime number or False otherwise
    // Examples
    // prime_length('Hello') == True
    // prime_length('abcdcba') == True
    // prime_length('kittens') == True
    // prime_length('orange') == False
    def primeLength(string : String) : Boolean = {
        if (string.length < 2) {
            return false
        }
        for (i <- 2 to sqrt(string.length).toInt) {
            if (string.length % i == 0) {
                return false
            }
        }
        return true

    }
    def main(args: Array[String]) = {
    assert(primeLength(("Hello")) == (true));
    assert(primeLength(("abcdcba")) == (true));
    assert(primeLength(("kittens")) == (true));
    assert(primeLength(("orange")) == (false));
    assert(primeLength(("wow")) == (true));
    assert(primeLength(("world")) == (true));
    assert(primeLength(("MadaM")) == (true));
    assert(primeLength(("Wow")) == (true));
    assert(primeLength(("")) == (false));
    assert(primeLength(("HI")) == (true));
    assert(primeLength(("go")) == (true));
    assert(primeLength(("gogo")) == (false));
    assert(primeLength(("aaaaaaaaaaaaaaa")) == (false));
    assert(primeLength(("Madam")) == (true));
    assert(primeLength(("M")) == (false));
    assert(primeLength(("0")) == (false));
    }

}
